### PR TITLE
add dlclassifier and unit test

### PR DESCRIPTION
**Pull request by hhbyyh**
## What changes were proposed in this pull request?
primary changes:
1. Add DLClassifier and DLClassifierModel for the classification tasks, in which the prediction column is Double type.
2. Add unit tests for different feature and label types. Array[Double], Array[Float] and Vector.
3. Just pass RDD[Array[_]] to internalFit or internalTransform to avoid creating the extra array column. 

## How was this patch tested?

added unit tests for DLEstimator and DLClassifier. Manual test for DLEstimatorLeNet example.

### CODE REVIEW

**Comment by hhbyyh** [comment]
PR validation passed: http://172.168.2.101:8080/job/BigDL-PR-Validation/1096/

File: spark/dl/src/main/scala/org/apache/spark/ml/DLClassifier.scala [hunk 1]
```diff
@@ -15,147 +15,74 @@
...
-  def setFeaturesCol(featuresColName: String): this.type = set(featuresCol, featuresColName)
-
-  def setPredictionCol(value: String): this.type = set(predictionCol, value)
+class DLClassifier[@specialized(Float, Double) T: ClassTag](
+    override val model: Module[T],
+    override val criterion : Criterion[T],
+    override val featureSize : Array[Int],
+    override val uid: String = "DLClassifier"
+  )(implicit ev: TensorNumeric[T])
+  extends DLEstimator[T](model, criterion, featureSize, Array(1)) with DLParams with HasBatchSize {
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/org/apache/spark/ml/DLClassifier.scala]: Do we need to explicitly add "with DLParams with HasBatchSize" as general class DLEstimator already does ?

File: spark/dl/src/main/scala/org/apache/spark/ml/DLEstimator.scala [hunk 2]
```diff
@@ -61,77 +66,61 @@ class DLEstimator[@specialized(Float, Double) T: ClassTag](
 
   def setBatchSize(value: Int): this.type = set(batchSize, value)
 
+  /**
+   * number of max Epoch for the training
+   */
   val maxEpoch = new IntParam(this, "maxEpoch", "number of max Epoch", ParamValidators.gt(0))
-  setDefault(maxEpoch -> 20)
+  setDefault(maxEpoch -> 100)
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/org/apache/spark/ml/DLEstimator.scala]: Just curious, why 20 to 200 ? empirical value ?

File: spark/dl/src/main/scala/org/apache/spark/ml/DLClassifier.scala (hunk 1 above)
Comment (hhbyyh) [inline spark/dl/src/main/scala/org/apache/spark/ml/DLClassifier.scala]: Good idea.

**Comment by hhbyyh** [comment]
Close the PR as I merged wrong git branch and break the history... I'll send a new one.

**Pull request closed without merging**